### PR TITLE
MWPW-141560: unav unit tests

### PR DESCRIPTION
**Pull request by robert-bogos**
## Description
This PR adds unit tests for the universal navigation

## Related Issue
Resolves: [MWPW-141560](https://jira.corp.adobe.com/browse/MWPW-141560)

## Testing instructions
1. change `web-test-runner.config.mjs` by adding   `files: ['./test/blocks/global-navigation/global-navigation.test.js']`, to the default exported json (this will help target the global-navigation.test.js for tests)
2. run npm run test:file:watch 

## Screenshots:
**Before:**
Screenshot 2024-02-07 at 16 43 17

**After:**
Screenshot 2024-02-07 at 16 41 18

## Test URLs
**Milo:**
- Before: https://gnav--milo--adobecom.hlx.page/drafts/rbogos/unav?martech=off
- After: https://mwpw-141560-unav-unit-tests--milo--robert-bogos.hlx.page/drafts/rbogos/unav?martech=off&imsClientId=fedsmilo

### CODE REVIEW

**Review by narcis-radu — APPROVED** [review]

File: libs/scripts/scripts.js [hunk 1]
```diff
@@ -123,7 +123,8 @@ const config = {
   geoRouting: 'on',
   fallbackRouting: 'on',
   links: 'on',
-  imsClientId: 'milo',
+  // will undo before merging
+  imsClientId: new URLSearchParams(window.location.search).get('imsClientId') || 'milo', // TODO: reset to original value
```
Comment (mokimo) [inline libs/scripts/scripts.js:127]: We should either keep this or start adding a few branches to the milo token.

File: test/blocks/global-navigation/global-navigation.test.js [hunk 2]
```diff
@@ -1161,4 +1199,164 @@ describe('global navigation', () => {
...
+      sinon.restore();
+      window.allow = orgAlloy;
+      if (document.head.contains(unavMeta)) document.head.removeChild(unavMeta);
+    });
+
+    describe('desktop', () => {
+      it('should render the Universal navigation', async () => {
+        const fullUnavMeta = toFragment`<meta name="universal-nav" content="profile, appswitcher, notifications, help">`;
+        const gnav = await createFullGlobalNavigation();
+        await gnav.decorateUniversalNav();
```
Comment (mokimo) [inline test/blocks/global-navigation/global-navigation.test.js:1226]: Should this be moved to the `createFullGlobalNavigation` helper, with a param to trigger decorating the universal nav or the old one?

File: test/blocks/global-navigation/global-navigation.test.js [hunk 3]
```diff
@@ -1161,4 +1199,164 @@ describe('global navigation', () => {
...
+        const unavFirstCallItems = window.UniversalNav.getCall(0).args[0]?.children;
+
+        expect(unavFirstCallItems[0]?.name === 'profile' && !unavFirstCallItems[1]).to.be.true;
+
+        document.head.removeChild(unavMeta);
+        document.head.append(fullUnavMeta);
+        await gnav.decorateUniversalNav();
+        const unavSecondCallItems = window.UniversalNav.getCall(1).args[0]?.children;
+
+        expect(unavSecondCallItems.every((c) => ['profile', 'app-switcher', 'notifications', 'help'].includes(c.name)))
```
Comment (mokimo) [inline test/blocks/global-navigation/global-navigation.test.js:1236]: Should we rather verify, the elements have been added to the DOM?

File: test/blocks/global-navigation/global-navigation.test.js [hunk 4]
```diff
@@ -1161,4 +1199,164 @@ describe('global navigation', () => {
...
+          .to.be.true;
+
+        document.head.removeChild(fullUnavMeta);
+      });
+
+      it('should reload unav on viewport change', async () => {
+        window.UniversalNav.reload = sinon.spy();
+        isDesktop.dispatchEvent(new Event('change'));
+        expect(window.UniversalNav.reload.getCall(0)).to.exist;
```
Comment (mokimo) [inline test/blocks/global-navigation/global-navigation.test.js:1245]: If this is just called once after the change event, should we verify it with `window.UniversalNav.reload.calledOnce` ?

**Review by mokimo — APPROVED** [review]

File: libs/scripts/scripts.js (hunk 1 above)
Comment (overmyheadandbody) [reply libs/scripts/scripts.js:127]: Yeah, this should be removed, I used it for a another branch for testing purposes, but as the TODO says, it shouldn't be merged in

File: test/blocks/global-navigation/global-navigation.test.js (hunk 2 above)
Comment (overmyheadandbody) [reply test/blocks/global-navigation/global-navigation.test.js:1271]: Agree on this one too, we might need to decorate the UNav in other tests too, let's have it available anywhere through `createFullGlobalNavigation`

File: test/blocks/global-navigation/global-navigation.test.js [hunk 5]
```diff
@@ -1208,4 +1244,164 @@ describe('global navigation', () => {
...
+
+        expect(unavSecondCallItems.every((c) => ['profile', 'app-switcher', 'notifications', 'help'].includes(c.name)))
+          .to.be.true;
+
+        document.head.removeChild(fullUnavMeta);
+      });
+
+      it('should reload unav on viewport change', async () => {
+        window.UniversalNav.reload = sinon.spy();
+        isDesktop.dispatchEvent(new Event('change'));
```
Comment (overmyheadandbody) [inline test/blocks/global-navigation/global-navigation.test.js:1289]: Is there no `.matches` condition in the logic? I was thinking we could actually resize the window to desktop/mobile and check that UNav is re-initialised.

**Review by overmyheadandbody — APPROVED** [review]

File: test/blocks/global-navigation/global-navigation.test.js (hunk 3 above)
Comment (narcis-radu) [reply test/blocks/global-navigation/global-navigation.test.js:1281]: The notifications are not added unless you are logged-in, so everything depends on the Universal Navigation. I don't think we need to check for the elements, just that the correct children have been passed.

Comment (robert-bogos) [reply test/blocks/global-navigation/global-navigation.test.js:1281]: Yes, that would fall under the responsibility of the Universal Navigation, we just need to make sure we are sending the correct children

File: test/blocks/global-navigation/global-navigation.test.js (hunk 4 above)
Comment (robert-bogos) [reply test/blocks/global-navigation/global-navigation.test.js:1290]: Good suggestion 👍 Made the change

File: test/blocks/global-navigation/global-navigation.test.js (hunk 5 above)
Comment (robert-bogos) [reply test/blocks/global-navigation/global-navigation.test.js:1289]: I did change to resizing the window with `await setViewport(viewports.mobile);`. Not sure where `.matches` could fit in here 🤔

File: libs/scripts/scripts.js [hunk 6]
```diff
@@ -123,6 +123,7 @@ const config = {
   geoRouting: 'on',
   fallbackRouting: 'on',
   links: 'on',
+  // will undo before merging
```
Comment (mokimo) [inline libs/scripts/scripts.js:126]: There's still a leftover comment in here

Comment (robert-bogos) [reply libs/scripts/scripts.js:126]: Thanks 👍

File: test/blocks/global-navigation/global-navigation.test.js (hunk 2 above)
Comment (robert-bogos) [reply test/blocks/global-navigation/global-navigation.test.js]: Good suggestion! Made the change 👍

**Comment by robert-bogos** [comment]
This can be merged now @narcis-radu @overmyheadandbody @mokimo

**Pull request merged**